assets/js/phoenix_live_view.js

```diff
@@ -704,6 +705,22 @@ let DOM = {
       onBeforeElUpdated: function(fromEl, toEl) {
         if(fromEl.isEqualNode(toEl)){ return false } // Skip subtree if both elems and children are equal
 
+        if (fromEl.type === "number") {
+          // We need to treat number inputs differently. Chrome will clear values if
+          // report bad inputs so we want to do nothing
+          // eg. 1..
+          if (fromEl.validity && fromEl.validity.badInput) {
+            return false
+          }
+
+          // Moreover, Firefox will delete the separator when going from `1.1` to `1.`
+          // So we should never update the input if it is the currently active element
+          // This is the "safe" route and approaches other communities like React have taken
+          if (fromEl.ownerDocument.activeElement === fromEl) {
```

Comment:
can we use the same `focused` check that we use elsewhere and that is in scope here?